shm/Island.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <ostream>
+
+#include "Coordinates.hpp"
+
+class Island
+{
+public:
+    Island();
+    Island(int x, int y);
+
+    Coordinates getPosition() { return position_; }
+
+private:
+    Coordinates position_;
+};
+
+bool operator==(Island& lhs, Island& rhs);
```

Comment:
Why it's not a part of Island class? Could it be defined as static method inside the class?